### PR TITLE
[IMAGING-259]: Coverage improvements, speed improvement for LZW/RGB

### PR DESCRIPTION
**Pull request by gwlucastrig**
Checking the Coverall reports, I spotted an unused area in DataReaderStrips and DataReaderTiled.  This block of code was attached to the 2012 performance fixes. As far as I can tell, there are no TIFF configurations that would exercise it. And, even if there were TIFF files that would use it, the new code in this pull request would still work on them.   The idea behind the old performance fixes was that we dedicate custom  blocks of code to frequently used settings and let the generic code handle the rest. Because the custom code could be tuned to the particular TIFF configuration, it would run faster.

I also noticed that the old code did not handle cases where RGB style files were compressed and used a predictor.  So I modified it slightly to handle the predictor rather than allowing the code to fall through to the generic block.  I tested this on a image I have that is quite large 6000-by-4000 pixels and the performance improved substantially.  Below are the results using the ApacheImagingSpeedAndMemory class that is in the commons-imaging example folder.

I tried to make the changes be as clear as possible to simplify inspection.  I will check Trevor later to make sure I managed to improve coverage (or at least not degrade it).

 ```
Processing file: CONUS_LandWaterMask_LZW_RGB.tif (original)
 image size: 6000 by 4000

 time to load image    --         memory
 time ms      avg ms   --    used mb   total mb
  971.817     0.000    --    213.592   252.000 
  921.690     0.000    --    143.229   260.000 
  895.587   895.587    --     96.234   174.000 
  899.227   897.407    --    117.259   154.000 
  899.078   897.964    --    134.200   184.000 
  889.602   895.873    --    143.226   180.000 
  896.170   895.933    --    128.183   188.000 
  894.250   895.652    --     97.187   178.000 
  896.436   895.764    --    103.226   186.000 
  891.540   895.236    --    119.185   171.000 
 
 Processing file: CONUS_LandWaterMask_LZW_RGB.tif (with chamges)
  image size: 6000 by 4000

 time to load image    --         memory
 time ms      avg ms   --    used mb   total mb
  498.123     0.000    --    212.589   252.000 
  423.136     0.000    --    110.733   237.000 
  396.021   396.021    --    100.735   164.000 
  400.435   398.228    --    115.725   160.000 
  400.901   399.119    --    114.726   162.000 
  395.092   398.112    --    118.711   159.000 
  394.106   397.311    --    118.710   159.000 
  400.866   397.903    --    118.710   159.000 
  400.972   398.342    --    115.710   160.000 
  397.218   398.201    --    109.691   164.000 
```

### CODE REVIEW

File: src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderTiled.java [hunk 1]
```diff
@@ -125,8 +125,8 @@ private void interpretTile(final ImageBuilder imageBuilder, final byte[] bytes,
         // verify that all samples are one byte in size
         final boolean allSamplesAreOneByte = isHomogenous(8);
 
-        if (predictor != 2 && bitsPerPixel == 24 && allSamplesAreOneByte) {
-            int k = 0;
+          if (bitsPerPixel == 24 && allSamplesAreOneByte
```
Comment (kinow) [inline src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderTiled.java:128]: I think this `if` is not aligned with the `final` statement above. Haven't looked at the code in the IDE, but maybe it is formatted with wrong number of spaces?

File: src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderStrips.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ private void interpretStrip(
         // verify that all samples are one byte in size
         final boolean allSamplesAreOneByte = isHomogenous(8);
 
-        if (predictor != 2 && bitsPerPixel == 8 && allSamplesAreOneByte) {
+         if (predictor != 2 && bitsPerPixel == 8 && allSamplesAreOneByte) {
```
Comment (kinow) [inline src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderStrips.java:147]: Misaligned with the lines above?

File: src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderTiled.java (hunk 1 above)
Comment (gwlucastrig) [reply src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderTiled.java:128]: The alignment is now fixed.

**Comment by gwlucastrig** [comment]
Do you want me to enter another JIRA item?   The performance enhancement does represent a functional change, but I didn't want to start spaming JIRA.

How do you want me to confirm performance testing?

**Comment by kinow** [comment]
> Do you want me to enter another JIRA item? The performance enhancement does represent a functional change, but I didn't want to start spaming JIRA.
> 
> How do you want me to confirm performance testing?

It is more because we are changing the code. Small changes like code style, typos, or adding new and simple/small tests can be added without a JIRA. But for changes in the code it's better to always have an issue associated, so that users/devs can link in discussions in the mailing list for example.

>How do you want me to confirm performance testing?

Sorry @gwlucastrig I thought you said you were going to measure the performance, but you actually said it was the coverage. :point_down: 

>I tried to make the changes be as clear as possible to simplify inspection. I will check Trevor later to make sure I managed to improve coverage (or at least not degrade it).

The bot has already confirmed that the coverage improved. So once we have the JIRA we are ready to merge this one and include in the changelog for the next release.

**Comment by gwlucastrig** [comment]
Enterred JIRA issue IMAGING-259 with a description of the changes.   I have no further code changes I plan to include in this Pull Request.  Please let me know if there is anything else you require.

**Pull request closed without merging**